tests/e2e/clusterreset/clusterreset_test.go

```diff
@@ -0,0 +1,181 @@
+var agentCount = flag.Int("agentCount", 2, "number of agent nodes")
+var hardened = flag.Bool("hardened", false, "true or false")
+var ci = flag.Bool("ci", false, "running on CI")
+var local = flag.Bool("local", false, "deploy a locally built K3s binary")
+
+// Environment Variables Info:
+// E2E_EXTERNAL_DB: mysql, postgres, etcd (default: etcd)
+// E2E_RELEASE_VERSION=v1.23.1+k3s2 (default: latest commit from master)
+
+func Test_E2EClusterReset(t *testing.T) {
+	RegisterFailHandler(Fail)
+	flag.Parse()
+	RunSpecs(t, "Create ClusterReset Test Suite")
+}
+
+var (
+	kubeConfigFile  string
+	serverNodeNames []string
+	agentNodeNames  []string
+)
+
+var _ = Describe("Verify Create", Ordered, func() {
+	Context("Cluster :", func() {
+		It("Starts up with no issues", func() {
+			var err error
+			if *local {
+				serverNodeNames, agentNodeNames, err = e2e.CreateLocalCluster(*nodeOS, *serverCount, *agentCount)
+			} else {
+				serverNodeNames, agentNodeNames, err = e2e.CreateCluster(*nodeOS, *serverCount, *agentCount)
+			}
+			Expect(err).NotTo(HaveOccurred(), e2e.GetVagrantLog())
+			fmt.Println("CLUSTER CONFIG")
+			fmt.Println("OS:", *nodeOS)
+			fmt.Println("Server Nodes:", serverNodeNames)
+			fmt.Println("Agent Nodes:", agentNodeNames)
+			kubeConfigFile, err = e2e.GenKubeConfigFile(serverNodeNames[0])
+			Expect(err).NotTo(HaveOccurred())
+			Expect(e2e.DockerLogin(kubeConfigFile, *ci)).To(Succeed())
+		})
+
+		It("Checks Node and Pod Status", func() {
+			fmt.Printf("\nFetching node status\n")
+			Eventually(func(g Gomega) {
+				nodes, err := e2e.ParseNodes(kubeConfigFile, false)
+				g.Expect(err).NotTo(HaveOccurred())
+				for _, node := range nodes {
+					g.Expect(node.Status).Should(Equal("Ready"))
+				}
+			}, "420s", "5s").Should(Succeed())
+			_, _ = e2e.ParseNodes(kubeConfigFile, true)
+
+			fmt.Printf("\nFetching Pods status\n")
+			Eventually(func(g Gomega) {
+				pods, err := e2e.ParsePods(kubeConfigFile, false)
+				g.Expect(err).NotTo(HaveOccurred())
+				for _, pod := range pods {
+					if strings.Contains(pod.Name, "helm-install") {
+						g.Expect(pod.Status).Should(Equal("Completed"), pod.Name)
+					} else {
+						g.Expect(pod.Status).Should(Equal("Running"), pod.Name)
+					}
+				}
+			}, "420s", "5s").Should(Succeed())
+			_, _ = e2e.ParsePods(kubeConfigFile, true)
+		})
+
+		It("Verifies ClusterReset Functionality", func() {
+			Eventually(func(g Gomega) {
+				for _, nodeName := range serverNodeNames {
+					if nodeName != "server-0" {
+						cmd := "sudo systemctl stop k3s"
+						_, err := e2e.RunCmdOnNode(cmd, nodeName)
+						Expect(err).NotTo(HaveOccurred())
+					}
+				}
+
+				cmd := "sudo systemctl stop k3s"
+				_, err := e2e.RunCmdOnNode(cmd, "server-0")
+				Expect(err).NotTo(HaveOccurred())
+
+				cmd = "sudo k3s server --cluster-reset"
+				res, err := e2e.RunCmdOnNode(cmd, "server-0")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(res).Should(ContainSubstring("cluster-reset"))
```

Comment:
Could this expect ever get false positives? I wonder if we change this to be most of that full last line from the cluster-reset command. I don't have it off the top of my head, but it's something like "cluster-reset has finished successfully. On existing nodes, please delete ${datadir]/db and rejoin"